File: Python/Arrays & Hashing/top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        nums_count = dict()
        for num in nums:
            nums_count[num] = nums_count.get(num, 0) + 1
        
        ret = sorted(nums_count.items(), key=lambda item: item[1], reverse=True)[:k]
        return [x[0] for x in ret[:k]]

File: Cpp/Arrays & Hashing/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> numsCount;
        vector<pair<int, int>> pairs;

        for (int num : nums) {
            numsCount[num]++;
        }

        vector<int> ret;
        for (auto item = numsCount.begin(); item != numsCount.end(); item++) {
            pairs.push_back({item->second, item->first});
        }

        sort(pairs.rbegin(), pairs.rend());

        for (int i = 0; i < k; i++) {
            ret.push_back(pairs[i].second);
        }

        return ret;
    }
};
